psl-core/src/main/java/org/linqs/psl/reasoner/dcd/DCDReasoner.java

```diff
@@ -74,7 +74,7 @@ public double optimize(TermStore baseTermStore) {
         boolean converged = false;
         int iteration = 1;
 
-        for (; iteration < (maxIterations * budget) && !converged; iteration++) {
+        while(!converged) {
```

Comment:
Use another name instead of `converged`, since it may not have converged.
(Same with SGD.)